plugins/hosts/windows/cap/ps.rb

```diff
@@ -0,0 +1,50 @@
+require "pathname"
+require "tmpdir"
+
+require "vagrant/util/subprocess"
+
+module VagrantPlugins
+  module HostWindows
+    module Cap
+      class PS
+        def self.ps_client(env, ps_info)
+          logger = Log4r::Logger.new("vagrant::hosts::windows")
+
+          command = <<-EOS
+            $plain_password = "#{ps_info[:password]}"
+            $username = "#{ps_info[:username]}"
+            $port = "#{ps_info[:port]}"
+            $hostname = "#{ps_info[:host]}"
+            $password = ConvertTo-SecureString $plain_password -asplaintext -force
+            $creds = New-Object System.Management.Automation.PSCredential ("$hostname\\$username", $password)
+            function prompt { kill $PID }
+            Enter-PSSession -ComputerName $hostname -Credential $creds -Port $port
+          EOS
+
+          logger.debug("Starting remote powershell with command:\n#{command}")
+          command = command.chars.to_a.join("\x00").chomp
+          command << "\x00" unless command[-1].eql? "\x00"
```

Comment:
This entire encoding block was pretty much a copy/paste from the winrm gem as it stood at the time of this PR. This has changed quite a bit since. I am going to replace all of this with:
```
command = ::WinRM::PowershellScript.new(command).encoded
```